packages/driver/src/cy/commands/navigation.js

```diff
@@ -785,29 +768,47 @@ module.exports = (Commands, Cypress, cy, state, config) => {
       const runnable = state('runnable')
 
       const changeIframeSrc = (url, event) => {
```

Comment:
Errors in cy.visit's onBeforeLoad used to be handled outside the command itself and so would trigger the global 'error' handler instead of failing cy.visit itself. This differed from the onLoad function, which is handled within the cy.visit command. It was also awkward because we had to look up the command to get to the onBeforeLoad function, but it's unnecessary when actually in the visit command's code because it gets passed in as an option.